Classes/Service/AssetService.php

```diff
@@ -151,4 +160,29 @@ protected function findOrCreateTag(): Tag
 
         return $tag;
     }
+
+    protected function getShopResourceLastModified(string $url): \DateTime
+    {
+        return empty($this->fetchShopResourceState($url)->getHeader('last-modified')) ? new \DateTime() : new \DateTime($this->fetchShopResourceState($url)->getHeader('last-modified')[0]);
+    }
+
+    protected function shopResourceExists(string $url): bool
+    {
+        if ($this->fetchShopResourceState($url)->getStatusCode() === 200) {
+            return true;
+        }
+
+        $this->logger->warning(sprintf('Resource %s was not found. HTTP status code: %s', $url, $statusCode), LogEnvironment::fromMethodName(__METHOD__));
```

Comment:
Here the variable `$statusCode` is not defined and therefore the logger call throws an exception